### PR TITLE
BPTL Dashboard: Add homecollection APIs (Part 2.3)

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
https://github.com/episphere/connect/issues/771
Related PR:
https://github.com/episphere/biospecimen/pull/622

Updated APIs to enhance functionality & fix errors

### CODE REVIEW

**Review by ramavavi — APPROVED** [review]

**Comment by ramavavi** [comment]
looks good.

**Review by Gbarra9 — APPROVED** [review]
Looks good. (concept id for BioKit_Mouthwash updated)

File: utils/firestore.js [hunk 1]
```diff
@@ -2100,7 +2100,7 @@ const addKitStatusToParticipant = async (participantsCID) => {
                     '266600170': {
                         ...prevParticipantObject,
                         '915179629': '664882224',
-                        '803510566': {
+                        '8583443674': {
                             '221592017': '849527480'
```
Comment (we-ai) [inline utils/firestore.js:2104]: 849527480 as number.

File: utils/firestore.js [hunk 2]
```diff
@@ -2073,7 +2073,7 @@ const queryTotalAddressesToPrint = async () => {
 
 const eligibleParticipantsForKitAssignment = async () => {
     try {
-        const snapshot = await db.collection("participants").where('173836415.266600170.803510566.221592017', '==', '849527480').get();
+        const snapshot = await db.collection("participants").where('173836415.266600170.8583443674.221592017', '==', '849527480').get();
```
Comment (we-ai) [inline utils/firestore.js:2076]: 849527480 as number

File: utils/firestore.js [hunk 3]
```diff
@@ -2171,7 +2171,7 @@ const assignKitToParticipant = async (data) => {
                 '266600170': {
                     ...prevParticipantObject,
                     '915179629': '534621077',
-                    '803510566': {
+                    '8583443674': {
                         '379252329': '390351864', // mouthwash
                         '221592017': '241974920',
```
Comment (we-ai) [inline utils/firestore.js:2176]: 390351864 and 241974920 as numbers

File: utils/firestore.js [hunk 4]
```diff
@@ -2235,7 +2235,7 @@ const confirmShipmentKit = async (shipmentData) => {
         const updatedParticipantObject = {
             '173836415': {
                 '266600170': {
-                    '803510566': {
+                    '8583443674': {
                         ...prevParticipantObject,
                         '221592017': '277438316',
```
Comment (we-ai) [inline utils/firestore.js:2240]: 277438316 as number

File: utils/firestore.js [hunk 5]
```diff
@@ -2302,11 +2305,11 @@ const storeKitReceipt = async (package) => {
...
-            '254109640': '353358909',
-            '173836415.266600170.915179629': '103209024',
+            '684635302': 353358909,
+            '254109640': 353358909,
+            '173836415.266600170.915179629': 103209024,
             '173836415.266600170.448660695': package['678166505'],
-            '173836415.266600170.803510566': {
+            '173836415.266600170.8583443674': {
                 ...prevParticipantObject,
                 '221592017': '375535639',
```
Comment (we-ai) [inline utils/firestore.js:2314]: number

**Review by we-ai — CHANGES_REQUESTED** [review]
Make sure concept IDs are numbers when used as values, otherwise downstream logics may/will break.

**Review by we-ai — APPROVED** [review]
Looks good.

**Review by jhflorey — APPROVED** [review]
look good

**Review by jhflorey — APPROVED** [review]

**Pull request merged**